### PR TITLE
Allow concurrent instances of GCSContainer

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

**Review by dimas-b — DISMISSED** [review]

File: testing/gcs-container/src/main/java/org/projectnessie/testing/gcs/GCSContainer.java [hunk 1]
```diff
@@ -97,9 +100,16 @@ public GCSContainer() {
   public GCSContainer(String image) {
     super(image == null ? DEFAULT_IMAGE + ":" + DEFAULT_TAG : image);
 
+    ThreadLocalRandom rand = ThreadLocalRandom.current();
+    localAddress =
```
Comment (adutra) [inline testing/gcs-container/src/main/java/org/projectnessie/testing/gcs/GCSContainer.java:104]: I'm pretty sure this won't work on MacOS without fiddling with `ifconfig`, let me test that and get back to you.

Why not pick up a random available port and use it? Does the container absolutely requires 4443 and nothing else?

Comment (adutra) [reply testing/gcs-container/src/main/java/org/projectnessie/testing/gcs/GCSContainer.java:104]: Confirmed :-(

```
java.lang.RuntimeException: com.google.cloud.storage.StorageException: Connect timed out
	at org.projectnessie.testing.gcs.GCSContainer.start(GCSContainer.java:138)
	at org.projectnessie.gc.iceberg.inttest.ITSparkIcebergNessieGCP.setupGcs(ITSparkIcebergNessieGCP.java:38)
```

Can you at least add an `if(macOs) { ... }` branch?

Comment (snazy) [reply testing/gcs-container/src/main/java/org/projectnessie/testing/gcs/GCSContainer.java:104]: Ah, right. There was that behavior difference :()

Comment (snazy) [reply testing/gcs-container/src/main/java/org/projectnessie/testing/gcs/GCSContainer.java:106]: Fun fact: now I recall that I had a script for macOS in the past that added a ton of additional `127.x.y.z` addresses.

**Comment by snazy** [comment]
Please wait a bit before approving this one. Maybe I can get it working w/ networks in some way.

**Comment by snazy** [comment]
> Please wait a bit before approving this one. Maybe I can get it working w/ networks in some way.

Bummer - not as easy as I thought :(

**Review by adutra — APPROVED** [review]

**Pull request merged**